garbage-collection/gc_cleanup.py

```diff
@@ -0,0 +1,148 @@
+# argo garbage collection
+
+import os
+import argparse
+import json, yaml
+from datetime import datetime, tzinfo, timedelta
+import logging
+
+
+from kubernetes import client, config, watch
+from kubernetes.client.rest import ApiException
+from pprint import pprint
+
+
+# config.load_incluster_config()
+
+config.load_kube_config()
+
+api_client = client.ApiClient()
+custom_api = client.CustomObjectsApi(api_client)
+v1_api = client.CoreV1Api(api_client)
+
+def get_pods(workflow,namespace):
+    # check if pods exist in workflow and collect them
+    nodes = list(workflow['status']['nodes'].keys())
+    pods = []
+    for node in nodes:
+        try:
+            api_response = v1_api.read_namespaced_pod_status(node, namespace)
+            pods.append(node)
+        except ApiException as e:
+            # print("Exception when calling CoreV1Api->read_namespaced_pod: %s\n" % e)
+            pass
+    return pods
+
+
+def delete_pods(pod,namespace,body):
+    try:
+        api_response = v1_api.delete_namespaced_pod(pod, namespace, body=body, propagation_policy='Background')
+    except ApiException as e:
+        print("Exception when calling CoreV1Api->delete_namespaced_pod: %s\n" % e)
```

Comment:
if exception is called, api response will not be initiated. Return will raise NameError. Also print that says workflow is deleted will be misleading